### PR TITLE
Added traverse and foldable instance for NonEmptyList

### LINKED ISSUE #202: Add missing traverse instance for NonEmptyList

### PR DESCRIPTION
**Pull request by tonilopezmr**
Fixes  #202

Added `Foldable` and `Traverse` for NonEmptyList.
Added tests for `TraverseLaws`.

### CODE REVIEW

File: kategory/src/main/kotlin/kategory/data/NonEmptyList.kt [hunk 1]
```diff
@@ -60,7 +60,11 @@ package kategory
...
         fun semigroupK(): SemigroupK<NonEmptyListHK> = object : NonEmptyListSemigroupK {}
+
+        fun foldable(): Foldable<NonEmptyListHK> = this
+
+        fun traverse(): Traverse<NonEmptyListHK> = this
     }
 }
 
-fun <A> A.nel(): NonEmptyList<A> = NonEmptyList.of(this)
+fun <A> A.k(): NonEmptyList<A> = NonEmptyList.of(this)
```
Comment (pakoito) [inline kategory/src/main/kotlin/kategory/data/NonEmptyList.kt]: I'd personally revert this change. k() is reserved for kind wrappers for list or function. In this case you're creating a nel() from a, the same way we have left() and right() for Either, success() for Try or valid() for Validated.

File: kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt [hunk 2]
```diff
@@ -32,6 +34,23 @@ interface NonEmptyListInstances :
         }
     }
 
+    override fun <A, B> foldL(fa: HK<NonEmptyListHK, A>, b: B, f: (B, A) -> B): B = fa.ev().tail.fold(f(b, fa.ev().head), f)
+
+    override fun <A, B> foldR(fa: HK<NonEmptyListHK, A>, lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> {
+        fun loop(fa_p: NonEmptyList<A>): Eval<B> = when {
```
Comment (pakoito) [inline kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt]: Is loop stack-safe here? I know in the past we've found differences in implementation from cats in this regard. I see ListKW has the same implementation, maybe it needs a law.

File: kategory/src/main/kotlin/kategory/data/NonEmptyList.kt (hunk 1 above)
Comment (tonilopezmr) [inline kategory/src/main/kotlin/kategory/data/NonEmptyList.kt]: Okey I'm going to revert 👍

File: kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt (hunk 2 above)
Comment (tonilopezmr) [inline kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt]: Yep, I have refactored `NonEmptyList.foldR` to use `ListKW.foldR`

Comment (pakoito) [inline kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt]: Yes, I mean, is ListKW.foldR stack-safe :D That's beyond the scope of this ticket. I'll approve as it is, we can investigate later.

Comment (tonilopezmr) [inline kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt]: 💯
